nextjs-app/components/pages/faq/ContentWithFilter.tsx

```diff
@@ -0,0 +1,27 @@
+"use client";
+
+import TagFilter from "@/components/common/TagFilter/TagFilter";
+import { FAQ_FILTER_OPTIONS } from "@/constants/filter-options";
+import { useState } from "react";
+
+const ContentWithFilter = () => {
+  const [selectedFilter, setSelectedFilter] = useState<string>("all");
```

Comment:
NIT: we don't need to specify the string type, ts will infer the type automatically
```suggestion
  const [selectedFilter, setSelectedFilter] = useState("all");
```